### PR TITLE
[ticket/13637] Add core.ucp_pm_compose_alter_vars PHP event

### PR DESCRIPTION
**Pull request by kinerity**
Add a new PHP event to ucp_pm_compose to alter the default vars

PHPBB3-13637

### CODE REVIEW

**Comment by VSEphpbb** [comment]
"alter_vars" is weird and not consistent with our event naming. I'd suggest `core.ucp_pm_compose_modify_data`

File: phpBB/includes/ucp/ucp_pm_compose.php [hunk 1]
```diff
@@ -90,6 +90,32 @@ function compose_pm($id, $mode, $action, $user_folders = array())
 	// we include the language file here
 	$user->add_lang('viewtopic');
 
+	/**
+	* Modify the default vars before composing a PM
+	*
+	* @event core.ucp_pm_compose_alter_vars
+	* @var	int		msg_id					topic_id in the page request
+	* @var	int		to_user_id				The id of whom the message is to
+	* @var	int		to_group_id				The id of the group whom the message is to
```
Comment (marc1706) [inline phpBB/includes/ucp/ucp_pm_compose.php]: whom shouldn't be needed in this case

File: phpBB/includes/ucp/ucp_pm_compose.php [hunk 2]
```diff
@@ -90,6 +90,32 @@ function compose_pm($id, $mode, $action, $user_folders = array())
...
+	* @var	bool	submit					Whether the user is sending the PM or not
```
Comment (marc1706) [inline phpBB/includes/ucp/ucp_pm_compose.php]: Whether the form has been submitted

File: phpBB/includes/ucp/ucp_pm_compose.php [hunk 3]
```diff
@@ -90,6 +90,32 @@ function compose_pm($id, $mode, $action, $user_folders = array())
 	// we include the language file here
 	$user->add_lang('viewtopic');
 
+	/**
+	* Modify the default vars before composing a PM
+	*
+	* @event core.ucp_pm_compose_modify_data
+	* @var	int		msg_id					topic_id in the page request
```
Comment (marc1706) [inline phpBB/includes/ucp/ucp_pm_compose.php]: It's actually the post_id and not the topic_id.

**Pull request merged**